pkg/policy/mapstate.go

```diff
@@ -27,3 +46,30 @@ type MapStateEntry struct {
 	// PolicyKey.
 	ProxyPort uint16
 }
+
+// DetermineAllowFromWorld determines whether world should be allowed to
+// communicate with the endpoint, based on legacy Cilium 1.0 behaviour. It
+// inserts the PolicyKey corresponding to the world in the desiredPolicyKeys
+// if the legacy mode is enabled.
+//
+// This must be run after DetermineAllowLocalhost().
+//
+// For more information, see https://cilium.link/host-vs-world
+func (keys MapState) DetermineAllowFromWorld() {
```

Comment:
When can we remove this? cc @joestringer

Reply:
Let's do it in a dedicated PR and not mix it into this cleanup